reasoner/resolution/resolver/ConcludableResolver.java

```diff
@@ -27,39 +28,69 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.Set;
 
-public class ConcludableResolver extends SubsumptiveCoordinator<ConcludableResolver, BoundConcludableResolver> {
+public class ConcludableResolver extends SubsumptiveCoordinator<ConcludableResolver> {
 
     private static final Logger LOG = LoggerFactory.getLogger(ConcludableResolver.class);
 
     private final LinkedHashMap<Driver<ConclusionResolver>, Set<Unifier>> conclusionResolvers;
     private final Concludable concludable;
+    private final Map<AnswerState.Partial.Concludable<?>, Driver<? extends BoundConcludableResolver<?>>> boundResolversByPartial;
```

Comment:
This map caches the cycle detection computation. This seems prudent as we may hit it very often. Looking at it now, it will grow to a large size, so perhaps this optimisation is unwarranted. Let's discuss.